lambdas/FetchJobsDataFiltered.py

```diff
@@ -0,0 +1,138 @@
+import psycopg2
+import psycopg2.extras
+import json
+import os
+import logging
+from typing import Dict, List, Any
+import datetime
+
+try:
+    # for local development, load environment variables from .env file
+    import dotenv
+    dotenv.load_dotenv()
+except ImportError:
+    logging.warning('dotenv not available, assuming running in AWS environment')
+    
+# Constants for environment variables
+DB_HOST = os.environ['DB_HOST']
+DB_PORT = os.environ['DB_PORT']
+DB_NAME = os.environ['DB_NAME']
+DB_USER = os.environ['DB_USER']
+DB_PASSWORD = os.environ['DB_PASSWORD']
+
+# Province mapping for Canadian provinces in a standardized format
+PROVINCE_MAPPING = {
+    "ab": "alberta",
+    "bc": "britishcolumbia",
+    "mb": "manitoba",
+    "nb": "newbrunswick",
+    "nl": "newfoundlandandlabrador",
+    "ns": "novascotia",
+    "on": "ontario",
+    "pe": "princeedwardisland",
+    "qc": "quebec",
+    "sk": "saskatchewan",
+    "nt": "northwestterritories",
+    "nu": "nunavut",
+    "yt": "yukon"
+}
+
+def standardize_location(location: str) -> str:
+    """Standardize the location string to lowercase without spaces."""
+    return location.lower().replace(" ", "")
+
+def get_db_connection():
+    """Establishes a database connection using configured environment variables."""
+    try:
+        return psycopg2.connect(
+            host=DB_HOST, port=DB_PORT, dbname=DB_NAME, 
+            user=DB_USER, password=DB_PASSWORD)
+    except Exception as e:
+        logging.error("Database connection failed: %s", e)
+        raise
+
+def get_date_days_ago(days: str) -> str:
+    """Calculates the date 'days' number of days ago from today."""
+    try:
+        days_ago = int(days)
+        target_date = datetime.datetime.now() - datetime.timedelta(days=days_ago)
+        return target_date.strftime('%Y-%m-%d')
+    except ValueError:
+        logging.error("Invalid input for days: %s", days)
```

Comment:
Logging need to be more specific. Look at the other comments about logging in this PR.

Imagine using AWS Cloud watch or some logging platform & all you see is error in production with 1 line message.

Ideally, you need, what lambda function is causing, what sub function is causing the error, blah blah, the more info the better. Also you should attach like an ID to an error, like an `errorID`. 

We can then query this error by id later for debugging in prod